### PR TITLE
[georeferencer] Support drag and dropping raster files onto the dialog

### PR DESCRIPTION
**Pull request by nirvn**
## Description

Title says it all. We've added drag and dropping of files on most QGIS windows/dialogs, let's not leave the georeferencer behind.

### CODE REVIEW

**Comment by nirvn** [comment]
This was requested via #34418 , which I was oblivious to until a minute ago.

File: src/app/georeferencer/qgsgeorefmainwindow.cpp [hunk 1]
```diff
@@ -231,37 +236,41 @@ void QgsGeoreferencerMainWindow::openRaster()
...
+    QString otherFiles = tr( "All other files (*)" );
+    QString lastUsedFilter = s.value( QStringLiteral( "/Plugin-GeoReferencer/lastusedfilter" ), otherFiles ).toString();
 
-  QString filters = QgsProviderRegistry::instance()->fileRasterFilters();
-  filters.prepend( otherFiles + ";;" );
-  filters.chop( otherFiles.size() + 2 );
-  mRasterFileName = QFileDialog::getOpenFileName( this, tr( "Open Raster" ), dir, filters, &lastUsedFilter, QFileDialog::HideNameFilterDetails );
-  mModifiedRasterFileName.clear();
+    QString filters = QgsProviderRegistry::instance()->fileRasterFilters();
+    filters.prepend( otherFiles + ";;" );
```
Comment (m-kuhn) [inline src/app/georeferencer/qgsgeorefmainwindow.cpp:249]:
```suggestion
    filters.prepend( otherFiles + QStringLiteral( ";;" ) );
```

File: src/app/georeferencer/qgsgeorefmainwindow.cpp [hunk 2]
```diff
@@ -296,6 +305,94 @@ void QgsGeoreferencerMainWindow::openRaster()
...
+  timer->setInterval( 50 );
+
+  // get the file list
+  QList<QUrl>::iterator i;
+  QList<QUrl>urls = event->mimeData()->urls();
+  QString file;
+  for ( i = urls.begin(); i != urls.end(); ++i )
+  {
+    QString fileName = i->toLocalFile();
+#ifdef Q_OS_MAC
```
Comment (m-kuhn) [inline src/app/georeferencer/qgsgeorefmainwindow.cpp:324]: Did you copy this from somewhere?
Should we put this into a utils file instead?

**Review by m-kuhn — APPROVED** [review]
Thank you

File: src/app/georeferencer/qgsgeorefmainwindow.cpp (hunk 2 above)
Comment (nirvn) [reply src/app/georeferencer/qgsgeorefmainwindow.cpp:324]: I took it from the layout designer (I assume qgisapp has the same code). +1 for code deduplication, but I'm not sure where we'd put this static function.

Comment (m-kuhn) [reply src/app/georeferencer/qgsgeorefmainwindow.cpp:324]: QgsGuiUtils?

Comment (nirvn) [reply src/app/georeferencer/qgsgeorefmainwindow.cpp:324]: @m-kuhn , actually, the underlying issue was fixed in Qt 5.4.1 (https://codereview.qt-project.org/c/qt/qtbase/+/103158) -- are we planning to support Qt5 pre 5.4.1 for QGIS 3.18? If not, I'd suggest we get rid of this hack altogether.

Comment (m-kuhn) [reply src/app/georeferencer/qgsgeorefmainwindow.cpp:324]: https://github.com/qgis/QGIS-Enhancement-Proposals/issues/197
+1 to get rid of it

Comment (nirvn) [reply src/app/georeferencer/qgsgeorefmainwindow.cpp]: @m-kuhn , done.

**Pull request merged**